non_profit/non_profit/doctype/member/member.py

```diff
@@ -11,9 +11,57 @@
 from payments.utils import get_payment_gateway_controller
 
 from non_profit.non_profit.doctype.membership_type.membership_type import get_membership_type
+from erpnext.accounts.doctype.subscription.subscription import Subscription
+from erpnext.accounts.doctype.subscription.subscription import is_prorate, get_prorata_factor, process_all
 
 
 class Member(Document):
+    @frappe.whitelist()
+    def create_membership_and_link(self):
+        if self.member_approval_status != "Approved":
+            frappe.msgprint(_("Member is not approved yet"))
+            return
+
+        existing_membership = frappe.get_all(
+            "Membership", filters={"member": self.name})
+        if existing_membership:
+            frappe.msgprint(_("Membership already exists for this Member"))
+            return
+
```

Comment:
The workflow should be:  
1. upon save of an approved member initiate Membership.(A member can have more than 1 membership i.e past years) Validatio on line 28 should be to only check if member has a current membership or not.
2. Dont restrict saving if member is not approved, i thing line 21 does this.